hydra-main/src/main/java/com/addthis/hydra/minion/Minion.java

```diff
@@ -246,6 +248,12 @@ private Minion(@JsonProperty("dataDir") File rootDir,
         } else {
             myHost = InetAddress.getLocalHost().getHostAddress();
         }
+        if (hostLocationInitializer == null) {
```

Comment:
If this could be null, you should also make the parameter in the constructor `@Nullable`